openhtf/__init__.py

```diff
@@ -367,20 +371,78 @@ def WithArgs(self, **kwargs):
     new_info.measurements = [m.WithArgs(**kwargs) for m in self.measurements]
     return new_info
 
-  def __call__(self, phase_data):
+  def __call__(self, test_state):
     kwargs = dict(self.extra_kwargs)
-    kwargs.update(phase_data.plug_manager.ProvidePlugs(
+    kwargs.update(test_state.plug_manager.ProvidePlugs(
         (plug.name, plug.cls) for plug in self.plugs if plug.update_kwargs))
     arg_info = inspect.getargspec(self.func)
-    # Pass in phase_data if it takes *args, or **kwargs with at least 1
+    # Pass in test_api if the phase takes *args, or **kwargs with at least 1
     # positional, or more positional args than we have keyword args.
     if arg_info.varargs or (arg_info.keywords and len(arg_info.args) >= 1) or (
         len(arg_info.args) > len(kwargs)):
-      # Underlying function has room for phase_data as an arg. If it doesn't
+      # Underlying function has room for test_api as an arg. If it doesn't
       # expect it but we miscounted args, we'll get another error farther down.
-      return self.func(phase_data, **kwargs)
+      return self.func(
+          test_state if self.options.requires_state else test_state.test_api,
+          **kwargs)
     return self.func(**kwargs)
 
 
+class TestApi(collections.namedtuple('TestApi', [
```

Comment:
hm, can you cite a source?  i couldn't find anything in either direction for python, but google c++ style dictates:
`Prefer to capitalize acronyms as single words (i.e. StartRpc(), not StartRPC()).`